menas/src/main/scala/za/co/absa/enceladus/menas/repositories/RunMongoRepository.scala

```diff
@@ -193,6 +193,94 @@ class RunMongoRepository @Autowired()(mongoDb: MongoDatabase)
       .aggregate[BsonDocument](pipeline)
   }
 
+  def getRunSummariesPerDatasetName(): Future[Seq[RunDatasetNameGroupedSummary]] = {
+    val pipeline = Seq(
+      project(fields(
+        include("dataset"),
+        Document("""{start: {
+                   |  $dateFromString: {
+                   |    dateString: "$startDateTime",
+                   |    format: "%d-%m-%Y %H:%M:%S %z"
+                   |  }
+                   |}},""".stripMargin),
+        Document(
+          """{timezone: {
+            |  $substrBytes: [
+            |    "$startDateTime", 20, 5
+            |  ]
+            |}}""".stripMargin),
+        excludeId()
+      )),
+      group("$dataset",
+        Accumulators.sum("numberOfRuns", 1),
+        Accumulators.max("latestStart", "$start"),
+        Accumulators.first("timezone", "$timezone")
+      ),
+      project(fields(
+        computed("datasetName", "$_id"),
+        include("numberOfRuns"),
+        Document("""{latestRunStartDateTime: {
+                   |  $dateToString: {
+                   |    date: "$latestStart",
+                   |    format: "%d-%m-%Y %H:%M:%S %z",
+                   |    timezone: "$timezone"
+                   |  }
+                   |}},""".stripMargin),
+        excludeId()
+      )),
+      sort(ascending("datasetName"))
+    )
+
+    collection
+      .aggregate[RunDatasetNameGroupedSummary](pipeline)
+      .toFuture()
+  }
+
+  def getRunSummariesPerDatasetVersion(datasetName: String): Future[Seq[RunDatasetVersionGroupedSummary]] = {
+    val pipeline = Seq(
+      filter(equal("dataset", datasetName)),
+      project(fields(
+        include("dataset", "datasetVersion"),
+        Document("""{start: {
+                   |  $dateFromString: {
+                   |    dateString: "$startDateTime",
+                   |    format: "%d-%m-%Y %H:%M:%S %z"
+                   |  }
+                   |}},""".stripMargin),
+        Document(
+          """{timezone: {
+            |  $substrBytes: [
+            |    "$startDateTime", 20, 5
+            |  ]
+            |}}""".stripMargin),
+        excludeId()
+      )),
+      group("$datasetVersion",
+        Accumulators.first("datasetName", "$dataset"),
+        Accumulators.sum("numberOfRuns", 1),
+        Accumulators.max("latestStart", "$start"),
+        Accumulators.first("timezone", "$timezone")
+      ),
+      project(fields(
+        computed("datasetVersion", "$_id"),
+        include("datasetName", "numberOfRuns"),
+        Document("""{latestRunStartDateTime: {
+                   |  $dateToString: {
+                   |    date: "$latestStart",
+                   |    format: "%d-%m-%Y %H:%M:%S %z",
+                   |    timezone: "$timezone"
+                   |  }
+                   |}},""".stripMargin),
+        excludeId()
+      )),
+      sort(ascending("datasetVersion"))
```

Comment:
I'm wondering if descending order here is more convenient?

Reply:
Makes sense